internal/services/artifact.go

```diff
@@ -74,3 +80,82 @@ func (s *artifactService) GetArtifactPolicies(ctx context.Context, artifact stri
 		},
 	}, nil
 }
+
+func (s *artifactService) GetImageMetadata(ctx context.Context, image string, username string, password string) (models.ImageMetadataResponse, error) {
+	ref, err := name.ParseReference(image)
+	if err != nil {
+		return models.ImageMetadataResponse{}, fmt.Errorf("invalid image URI: %w", err)
+	}
+
+	auth := authn.Anonymous
+	if username != "" && password != "" {
+		auth = &authn.Basic{Username: username, Password: password}
+	}
+
+	opts := []remote.Option{remote.WithAuth(auth), remote.WithContext(ctx)}
+
+	descriptor, err := remote.Get(ref, opts...)
+	if err != nil {
+		if isNotFound(err) {
+			return models.ImageMetadataResponse{}, fmt.Errorf("Image not found: %w", err)
```

Comment:
In go the convention is to use all lowercase strings, It may be worth setting up CI to check for bits like this https://github.com/securesign/secure-sign-operator/blob/main/.github/workflows/linter.yml and other things that may get missed :)